### PR TITLE
Drew branch

### PR DESCRIPTION
**Pull request by avwilliams1995**
Can now post with out a selected song or artist

### CODE REVIEW

File: app/(tabs)/explore.tsx [hunk 1]
```diff
@@ -1,82 +1,212 @@
...
-  const token: string = useSelector((state: RootState) => state.user.userToken);
+  const tempPosts: PostTypes[] = posts.sort((a, b) => {
+    const aScore = a.likes.length + a.comments.length;
+    const bScore = b.likes.length + b.comments.length;
+    return bScore - aScore;
+  });
+
+  const sortedPosts: PostTypes[] = tempPosts.filter(
+    (post) => post.albumCover !== ""
+  );
```
Comment (cyburns) [inline app/(tabs)/explore.tsx:33]: This is good for now but in production all posts could be massive which would take a long time. It would be best to write a hook that filters and sorts the data before fetching it from firebase

File: app/(tabs)/explore.tsx [hunk 2]
```diff
@@ -1,82 +1,212 @@
...
-                secondLine: `Playlist by: ${item.data.owner.display_name}`,
-              } || null
-            );
-        }
-      });
-      setSongs(items);
-    });
-  }
+  const toggleModal: ()=> void = () => {
+    console.log("toggledModal");
```
Comment (cyburns) [inline app/(tabs)/explore.tsx:38]: remove console logs

File: app/(tabs)/explore.tsx [hunk 3]
```diff
@@ -1,82 +1,212 @@
...
+          >
+            {!isModalVisible ? (<Pressable onPress={toggleModal}>
+              <View style={styles.searchBar}>
+                <Text style={{ color: Colors.lightGray, fontSize: 16 }}>
+                  Search for a user
+                </Text>
+              </View>
+              
+            </Pressable>) : <View style={styles.searchBarHidden}></View>}
+            <Modal
```
Comment (cyburns) [inline app/(tabs)/explore.tsx:62]: For me this modal is not in the same place as the search bar

File: app/(tabs)/explore.tsx [hunk 4]
```diff
@@ -1,82 +1,212 @@
...
+            >
+              Auda Top 10
+            </Text>
+            <ScrollView horizontal={true}>
+              {sortedPosts.slice(0, 10).map((post, index) => {
+                return (
+                  <View style={[styles.postCardContainer]}>
+                    <PostCard post={post} />
+
+                    <LinearGradient
```
Comment (cyburns) [inline app/(tabs)/explore.tsx:92]: gradient does not work with light mode and the black extends right of the image

File: app/(tabs)/explore.tsx [hunk 5]
```diff
@@ -1,82 +1,212 @@
...
-    alignItems: "center",
-    justifyContent: "flex-start",
     backgroundColor: Colors.black,
+    paddingLeft: 10,
+    paddingTop: 10,
+  },
+  titleText: {
+    color: Colors.white,
+    fontSize: 24,
+    fontWeight: "bold",
```
Comment (cyburns) [inline app/(tabs)/explore.tsx:127]: should be "mon-b"

File: app/(tabs)/index.tsx [hunk 6]
```diff
@@ -14,7 +14,7 @@ const HomePage = () => {
 
   const { isLoading, posts, refetchPosts } = getAllPosts();
 
-  const refreshPosts = async () => {
+   const refreshPosts = async () => {
```
Comment (cyburns) [inline app/(tabs)/index.tsx:17]: do you have prettier?

File: app/screens/create-post/createPost.tsx [hunk 7]
```diff
@@ -79,17 +82,25 @@ const CreatePost = () => {
...
     setIsLoading(true);
+    const spotifyUri = currentPostData !== null ? currentPostData.data.preview_url : null;
+
+
+    
+
+    
 
     try {
+      console.log(profilePicture,username, artistName, songName, albumCover, caption, userId, spotifyUri);
```
Comment (cyburns) [inline app/screens/create-post/createPost.tsx:103]: remove logs

File: app/screens/create-post/createPost.tsx [hunk 8]
```diff
@@ -102,8 +113,9 @@ const CreatePost = () => {
           likes: [],
           createdAt: new Date(),
           createdBy: userId,
-          spotifyUri: currentPostData.data.preview_url,
+          spotifyUri,
         };
+        console.log(newPostData);
```
Comment (cyburns) [inline app/screens/create-post/createPost.tsx:118]: remove logs

File: components/Posts/Post.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    color: Colors.white,
+    fontFamily: "mon-b",
+  },
+  mediumText: {
+    color: Colors.white,
+    fontFamily: "mon-med",
+  },
+});
+
+export default Post;
```
Comment (cyburns) [inline components/Posts/Post.tsx:93]: why create a whole new file so simlar to Posts.tsx instead of just resuing the component?

File: components/Posts/Posts.tsx [hunk 10]
```diff
@@ -63,7 +64,7 @@ const Posts = ({ posts }: PostsProps) => {
           {index === 3 && <SuggestedUsers />}
         </View>
       ))}
-      <View style={{ height: 50 }} />
+      {/* <View style={{ height: 50 }} /> */}
```
Comment (cyburns) [inline components/Posts/Posts.tsx:67]: Please uncomment this

File: components/Posts/SongCard.tsx [hunk 11]
```diff
@@ -107,7 +107,7 @@ const styles = StyleSheet.create({
   },
   postImage: {
     width: 95,
-    height: 95,
+    height: 100,
```
Comment (cyburns) [inline components/Posts/SongCard.tsx:110]: why did you change this? Album covers are square

File: components/exploreComponents/PostCard.tsx [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+          {post.spotifyUri !== null && (
+            <MyPlayer postId={post.id} preview_url={post.spotifyUri} />
+          )}
+        </View>
+      </View>
+      <View>
+        <Text style={styles.firstLine}>
+          {post.songName.length > 12
+            ? `${post.songName.substring(0, 12)}...`
+            : post.songName}
```
Comment (cyburns) [inline components/exploreComponents/PostCard.tsx:22]: Lets add the animated effect in here?

File: components/exploreComponents/PostCard.tsx [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+      </View>
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    display: "flex",
+    flexDirection: "column",
+    // backgroundColor: 'grey',
```
Comment (cyburns) [inline components/exploreComponents/PostCard.tsx:34]: remove

File: components/exploreComponents/TrendingAuda.tsx [hunk 14]
```diff
@@ -0,0 +1,182 @@
...
+                    style={{
+                      fontFamily: "mon-b",
+                      fontSize: 20,
+                      color: Colors.white,
+                      paddingLeft: 10,
+                    }}
+                  >
+                    {item.songName.length > 18
+                      ? item.songName.substring(0, 18) + "..."
+                      : item.songName}
```
Comment (cyburns) [inline components/exploreComponents/TrendingAuda.tsx:52]: Animated text here

File: components/exploreComponents/TrendingAuda.tsx [hunk 15]
```diff
@@ -0,0 +1,182 @@
+import { View, Text, TouchableOpacity, StyleSheet, Image } from "react-native";
```
Comment (cyburns) [inline components/exploreComponents/TrendingAuda.tsx:1]: Looks like this is the same component is Posts.tsx. Why make a whole new one?

File: components/searchComponents/SearchResult.tsx [hunk 16]
```diff
@@ -23,13 +23,14 @@ const SearchResult = ({
   secondLine,
   handlePress,
 }: SearchProps) => {
+  console.log(secondLine)
   return (
-    <TouchableOpacity onPress={() => handlePress(index)}>
+    <TouchableOpacity onPress={() => handlePress(index)} style= {{backgroundColor: Colors.black, opacity: .9}}>
```
Comment (cyburns) [inline components/searchComponents/SearchResult.tsx:28]: why do you change the opacity here and then again on line 48

File: components/searchComponents/UserSearch.tsx [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+      userId: selectedUser.id,
+    });
+    toggleModal();
+    setSearchInput("");
+    setSearchResults([]);
+  }
+
+  async function getItems() {
+    if (error) setError(false);
+    console.log("in get items", searchInput, allUsers[0]);
```
Comment (cyburns) [inline components/searchComponents/UserSearch.tsx:56]: remove log

File: components/searchComponents/UserSearch.tsx [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+    const users = allUsers.filter(function (user) {
+      return user.username.toLowerCase().indexOf(searchInput) != -1;
+    });
+    console.log(users);
```
Comment (cyburns) [inline components/searchComponents/UserSearch.tsx:60]: remove log

File: components/searchComponents/UserSearch.tsx [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+    users.length > 0 ? setSearchResults(users) : setError(true);
+  }
+
+  console.log(searchResults, isLoading);
```
Comment (cyburns) [inline components/searchComponents/UserSearch.tsx:64]: remove log

File: components/searchComponents/UserSearch.tsx [hunk 20]
```diff
@@ -0,0 +1,162 @@
+import {
```
Comment (cyburns) [inline components/searchComponents/UserSearch.tsx:1]: This component needs some styling upgrades overall but great work got a lot done today!

File: app/(tabs)/explore.tsx (hunk 1 above)
Comment (avwilliams1995) [reply app/(tabs)/explore.tsx:26]: I agree. Getting something back to the app for now, will look into this week

**Review by cyburns — APPROVED** [review]
